### ISSUE TITLE
System.DllNotFoundException for running in Linux

### ISSUE THREAD

**Issue by hmdhasani** [Dtronix/PDFiumCore#2]
I try to use pdfium to convert pdf to image in linux. but got this error:

```
Unhandled exception. System.DllNotFoundException: Unable to load shared library 'pdfium.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libpdfium.dll: cannot open shared object file: No such file or directory
   at PdfiumViewer.NativeMethods.Imports.FPDF_AddRef()
   at PdfiumViewer.NativeMethods.FPDF_AddRef()
   at PdfiumViewer.PdfLibrary..ctor()
   at PdfiumViewer.PdfLibrary.EnsureLoaded()
   at PdfiumViewer.PdfFile..ctor(Stream stream, String password)
   at PdfiumViewer.PdfDocument..ctor(Stream stream, String password)
   at PdfiumViewer.PdfDocument.Load(Stream stream, String password)
   at PdfiumViewer.PdfDocument.Load(String path, String password)
   at PdfiumViewer.PdfDocument.Load(String path)
   at PDFiumOnLinux.Program.Main(String[] args) in /src/Program.cs:line 10

```
This is my source code:

```
using PdfiumViewer;
using System.Drawing.Imaging;

namespace PDFiumOnLinux
{
    class Program
    {
        static void Main(string[] args)
        {
            using (var pdfDocument = PdfDocument.Load(@"Test.pdf"))
            {
                var bitmapImage = pdfDocument.Render(0, 300, 300, true);
                bitmapImage.Save(@"Test.jpg", ImageFormat.Jpeg);
            }
        }
    }
}
```

This is the csporj:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="PdfiumViewer" Version="2.13.0" />
    <PackageReference Include="PDFiumCore" Version="4503.0.0" />
    <PackageReference Include="System.Drawing.Common" Version="5.0.2" />
  </ItemGroup>

</Project>
```

and here's the docker-compose file:

```
version: "3.9"
services:
    app:
        image: mcr.microsoft.com/dotnet/sdk:3.1.409-buster
        volumes:
            - .:/src
        working_dir: /src
        entrypoint: bash -c "dotnet build PDFiumOnLinux.csproj && dotnet bin/Debug/netcoreapp3.1/PDFiumOnLinux.dll"

```

The program can be run using this command: 
`docker-compose run --rm app`

**Comment by DJGosnell** [comment]
Taking a stab in the dark here, but it appears you are using [PdfiumViewer's PdfDocument](https://github.com/pvginkel/PdfiumViewer/blob/master/PdfiumViewer/PdfDocument.cs) and not the PDFiumCore's libraries.

There is no interaction between PDFiumCore and PdfiumViewer as they are completely separate projects.  This project is just a direct C# <-> C++ Interop library.

I have a quick and dirty thread safe implementation written at https://github.com/Dtronix/DtronixPdf/

You can take a look at [RenderPageAction.cs](https://github.com/Dtronix/DtronixPdf/blob/master/src/DtronixPdf/Actions/RenderPageAction.cs) to see how rendering a page to a bitmap is handled.

**Comment by hmdhasani** [comment]
@DJGosnell  Thanks for your help. It saved me a lot.
The idea of adding PdfiumCore beside PdfiumViewer was from the fact that PdfiumViewer on windows does not work by itself and need another 'binary provider' nuget package that can be 'PdfiumViewer.Native.x86.v8-xfa' or 'PdfiumViewer.Native.x86_64.v8-xfa'.
So I was looking for a equivalent for Linux and thought PDFiumCore would be that. 
Anyway as PDFiumCore use a much newer version of pdfium, I decided to test it. So took the https://github.com/Dtronix/DtronixPdf/ and change RenderPageAction.cs to a PDFToImageConverter and run it on windows and linux and got the proper result!

I pushed the final code at https://github.com/Dtronix/DtronixPdf/. It's just a cleanup of DJGosnell's
